vslib/src/SwitchStateBaseMACsec.cpp

```diff
@@ -0,0 +1,859 @@
+        break;
+    }
+    return SAI_STATUS_FAILURE;
+}
+
+sai_status_t SwitchStateBase::loadMACsecAttr(
+    _In_ sai_object_type_t object_type,
+    _In_ sai_object_id_t object_id,
+    _Out_ MACsecAttr &macsec_attr)
+{
+    SWSS_LOG_ENTER();
+
+    std::vector<sai_attribute_t> attrs;
+    if (dumpObject(object_type, object_id, attrs))
+    {
+        if (loadMACsecAttr(
+                object_type,
+                object_id,
+                static_cast<uint32_t>(attrs.size()),
+                attrs.data(),
+                macsec_attr) != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_DEBUG(
+                "The %s %s is invalid",
+                sai_serialize_object_type(object_type).c_str(),
+                sai_serialize_object_id(object_id).c_str());
+            return SAI_STATUS_FAILURE;
+        }
+    }
+    else
+    {
+        SWSS_LOG_WARN(
+            "The %s %s is not existed",
+            sai_serialize_object_type(object_type).c_str(),
+            sai_serialize_object_id(object_id).c_str());
+        return SAI_STATUS_FAILURE;
+    }
+    return SAI_STATUS_SUCCESS;
+}
+
+sai_status_t SwitchStateBase::loadMACsecAttrsFromACLEntry(
+    _In_ sai_object_id_t entry_id,
+    _In_ const sai_attribute_t *entry_attr,
+    _In_ sai_object_type_t object_type,
+    _Out_ std::vector<MACsecAttr> &macsec_attrs)
+{
+    SWSS_LOG_ENTER();
+
+    if (entry_attr == nullptr || entry_id == SAI_NULL_OBJECT_ID)
+    {
+        SWSS_LOG_ERROR("entry is empty");
+        return SAI_STATUS_FAILURE;
+    }
+
+    if (entry_attr->id != SAI_ACL_ENTRY_ATTR_ACTION_MACSEC_FLOW)
+    {
+        SWSS_LOG_ERROR("Not include MACsec flow information");
+        return SAI_STATUS_FAILURE;
+    }
+    auto flow_id = entry_attr->value.aclaction.parameter.oid;
+
+    macsec_attrs.clear();
+    sai_attribute_t attr;
+
+    if (object_type == SAI_OBJECT_TYPE_MACSEC_PORT)
+    {
+        sai_object_id_t port_id = SAI_NULL_OBJECT_ID;
+        CHECK_STATUS(findPortByMACsecFlow(flow_id, port_id));
+        macsec_attrs.emplace_back();
+        if (findHostInterfaceInfoByPort(
+                port_id,
+                macsec_attrs.back().m_info) != SAI_STATUS_SUCCESS)
+        {
+            // The fail log has been record in findHostInterfaceInfoByPort
+            macsec_attrs.clear();
+            return SAI_STATUS_FAILURE;
+        }
+        macsec_attrs.back().m_veth_name =
+            vs_get_veth_name(macsec_attrs.back().m_info->m_name, port_id);
+        macsec_attrs.back().m_macsec_name =
+            SAI_VS_MACSEC_PREFIX + macsec_attrs.back().m_veth_name;
+        return SAI_STATUS_SUCCESS;
+    }
+
+    // Find all MACsec SCs that use this flow
+    std::vector<sai_object_id_t> macsec_scs;
+    attr.id = SAI_MACSEC_SC_ATTR_FLOW_ID;
+    attr.value.oid = flow_id;
+    findObjects(SAI_OBJECT_TYPE_MACSEC_SC, attr, macsec_scs);
+    if (object_type == SAI_OBJECT_TYPE_MACSEC_SC)
+    {
```

Comment:
I cannot move findObjects  into the `if`, because macsec_scs will be used at below code https://github.com/Azure/sonic-sairedis/pull/691/files?file-filters%5B%5D=.cpp#diff-254eea58a90a57d74818f0f06d67bcc09fa036c43af818c63988547bb9b99ae7R717